cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -1006,6 +1007,23 @@ public void grantKeyVaultAccessPolicyToServicePrincipal(String resourceGroupName
         });
     }
 
+    public boolean checkKeyVaultAccessPolicyForServicePrincipal(String resourceGroupName, String vaultName, String principalObjectId) {
```

Comment:
It would be nice to have an UT for this, at least for the logic after the `accessPolicies` initialization. `Azure` is `final` so cannot be mocked easily, thus most of `AzureClient` remains yet untested. Perhaps extract the
```
if (accessPolicies != null) {}
return false;
```
part into a method that has default (package-private) visibility, annotate it with `@VisibleForTesting` and add an UT for that method.

Note: I have not checked how feasible this is, in particular if all the types involved are indeed visible / can be stubbed (if necessary). If there was any such issue, of course, I would happily disregard this request.

CC @horadla23 @sodre90 I would be interested in your opinion on this as well.

Reply:
Added the UT for checking the permission snippet. As AccessPolicy is an interface, I have modified the code to check the individual AccessPolicyEntries for permissions.

Reply:
Yes, it is an interface. Is that a particular issue in terms of testing? Mockito loves to create mocks for interfaces, or even any specific dummy test implementation of that interface could be also added to the test class if some real-like logic is necessary.